streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -773,10 +775,32 @@
      * @param named  a {@link Named} config used to name the processor in the topology
      * @param predicates the ordered list of {@link Predicate} instances
      * @return multiple distinct substreams of this {@code KStream}
+     * @deprecated since 2.7. Use {@link #split(Named)} instead.
      */
+    @Deprecated
     @SuppressWarnings("unchecked")
     KStream<K, V>[] branch(final Named named, final Predicate<? super K, ? super V>... predicates);
 
+    /**
+     * Splits this stream. {@link BranchedKStream} can be used for routing the records to different branches depending
```

Comment:
Nit `Split` (no `s`) -- we use imperative to write JavaDocs.

`this stream` -> `this {@code KStream}.

(Same for the overload method)